### PR TITLE
Add maxIntersections to docs

### PR DESCRIPTION
**Pull request by rfraposa**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

File: docs/en/sql-reference/aggregate-functions/reference/maxintersections.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+```sql
+SELECT maxIntersections(start, end) FROM my_events;
+```
+
+Response:
+```response
+3
+```
+
+If you have multiple occurrences of the maximum interval, you can use the [`maxIntersectionsPosition` function](./maxintersectionsposition.md) to locate the numer and location of those occurrences.
```
Comment (DanRoscigno) [inline docs/en/sql-reference/aggregate-functions/reference/maxintersections.md:64]:
```suggestion
If you have multiple occurrences of the maximum interval, you can use the [`maxIntersectionsPosition` function](./maxintersectionsposition.md) to locate the number and location of those occurrences.
```

**Review by DanRoscigno — APPROVED** [review]
LGTM

File: docs/en/sql-reference/aggregate-functions/reference/maxintersections.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+The intervals look like the following:
+
+```response
+1 - 3
+1 - - - - 6
+  2 - - 5
+    3 - - - 7
+```
+
+Notice that three of these intervals have the value 4 in common, and that is the most frequent number of intersections:
```
Comment (qoega) [inline docs/en/sql-reference/aggregate-functions/reference/maxintersections.md:53]: It is really not clear why answer is not 4 with '3' as intersection. If we do not include boundaries it is important to explain.

Comment (rfraposa) [reply docs/en/sql-reference/aggregate-functions/reference/maxintersections.md:53]: There are not many aspects of this function that are "clear". I can only assume that it does not include boundaries - but let me play around with it, verify, and then update the docs.

File: docs/en/sql-reference/aggregate-functions/reference/maxintersections.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+The intervals look like the following:
+
+```response
+1 - 3
+1 - - - - 6
+  2 - - 5
+    3 - - - 7
+```
+
+Three of these intervals have the value 4 in common, and that is the most frequent number of intersections. (The intervals `(1,3)` and `(3,7)` share an endpoint but are not considered intersecting by the `maxIntersections` function.)
```
Comment (DanRoscigno) [inline docs/en/sql-reference/aggregate-functions/reference/maxintersections.md:53]:
```suggestion
Three of these intervals have a common value (the value is `4`, but the value that is common is not important, we are measuring the count of the intersections) . (The intervals `(1,3)` and `(3,7)` share an endpoint but are not considered intersecting by the `maxIntersections` function.)
```

Comment (DanRoscigno) [reply docs/en/sql-reference/aggregate-functions/reference/maxintersections.md:53]: This is a confusing function, I __think__ that what I wrote above is true.

Comment (rfraposa) [reply docs/en/sql-reference/aggregate-functions/reference/maxintersections.md]: LGTM.

**Pull request merged**